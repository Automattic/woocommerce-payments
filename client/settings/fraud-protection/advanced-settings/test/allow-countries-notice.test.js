/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';
/**
 * Internal dependencies
 */
import AllowedCountriesNotice from '../allow-countries-notice';
import FraudPreventionSettingsContext from '../context';

const mockContext = {
	advancedFraudProtectionSettings: {
		test_key: {
			enabled: false,
			block: false,
		},
	},
	setAdvancedFraudProtectionSettings: jest.fn(),
	setProtectionSettingsChanged: jest.fn(),
};

describe( 'Allowed countries rule card notice tests', () => {
	beforeAll( () => {
		global.wcSettings = {
			admin: {
				preloadSettings: {
					general: {
						woocommerce_allowed_countries: 'all',
						woocommerce_all_except_countries: [],
						woocommerce_specific_allowed_countries: [],
					},
				},
			},
			countries: {
				CA: 'Canada',
				US: 'United States',
			},
		};
	} );
	test( 'renders correctly when all countries are allowed', () => {
		const container = render(
			<FraudPreventionSettingsContext.Provider value={ mockContext }>
				<AllowedCountriesNotice setting={ 'test_key' } />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container.baseElement ).toHaveTextContent(
			/Enabling this filter will not have any effect because you are selling to all countries\./i
		);
	} );
	test( 'renders correctly when specific countries are allowed, others will be hold', () => {
		global.wcSettings.admin.preloadSettings.general.woocommerce_allowed_countries =
			'specific';
		global.wcSettings.admin.preloadSettings.general.woocommerce_specific_allowed_countries = [
			'CA',
			'US',
		];
		mockContext.advancedFraudProtectionSettings.test_key.block = false;

		const container = render(
			<FraudPreventionSettingsContext.Provider value={ mockContext }>
				<AllowedCountriesNotice setting={ 'test_key' } />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container.baseElement ).toHaveTextContent(
			/Orders outside from these countries will be screened by the filter: Canada, United States/i
		);
	} );
	test( 'renders correctly when specific countries are allowed, others will be blocked', () => {
		global.wcSettings.admin.preloadSettings.general.woocommerce_allowed_countries =
			'specific';
		global.wcSettings.admin.preloadSettings.general.woocommerce_specific_allowed_countries = [
			'CA',
			'US',
		];
		mockContext.advancedFraudProtectionSettings.test_key.block = true;

		const container = render(
			<FraudPreventionSettingsContext.Provider value={ mockContext }>
				<AllowedCountriesNotice setting={ 'test_key' } />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container.baseElement ).toHaveTextContent(
			/Orders outside from these countries will be blocked by the filter: Canada, United States/i
		);
	} );
	test( 'renders correctly when countries except some are allowed, others will be hold', () => {
		global.wcSettings.admin.preloadSettings.general.woocommerce_allowed_countries =
			'all_except';
		global.wcSettings.admin.preloadSettings.general.woocommerce_all_except_countries = [
			'CA',
			'US',
		];
		mockContext.advancedFraudProtectionSettings.test_key.block = false;

		const container = render(
			<FraudPreventionSettingsContext.Provider value={ mockContext }>
				<AllowedCountriesNotice setting={ 'test_key' } />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container.baseElement ).toHaveTextContent(
			/Orders from these countries will be screened by the filter: Canada, United States/i
		);
	} );
	test( 'renders correctly when countries except some are allowed, others will be blocked', () => {
		global.wcSettings.admin.preloadSettings.general.woocommerce_allowed_countries =
			'all_except';
		global.wcSettings.admin.preloadSettings.general.woocommerce_all_except_countries = [
			'CA',
			'US',
		];
		mockContext.advancedFraudProtectionSettings.test_key.block = true;

		const container = render(
			<FraudPreventionSettingsContext.Provider value={ mockContext }>
				<AllowedCountriesNotice setting={ 'test_key' } />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
		expect( container.baseElement ).toHaveTextContent(
			/Orders from these countries will be blocked by the filter: Canada, United States/i
		);
	} );
} );
