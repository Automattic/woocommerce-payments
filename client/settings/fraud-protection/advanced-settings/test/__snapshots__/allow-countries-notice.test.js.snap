// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allowed countries rule card notice tests renders correctly when all countries are allowed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
                Enabling this filter will not have any effect because you are selling to all countries.  
    </div>
    <div>
      <div
        class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
      >
        <div
          class="components-notice__content"
        >
          <div
            class="components-flex css-1nh0b0-Flex eboqfv50"
          >
            <div
              class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
            >
              <svg
                class="gridicon gridicons-notice-outline"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                  />
                </g>
              </svg>
            </div>
            <div
              class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
            >
              Enabling this filter will not have any effect because you are selling to all countries.
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
    >
      <div
        class="components-notice__content"
      >
        <div
          class="components-flex css-1nh0b0-Flex eboqfv50"
        >
          <div
            class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
          >
            <svg
              class="gridicon gridicons-notice-outline"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                />
              </g>
            </svg>
          </div>
          <div
            class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
          >
            Enabling this filter will not have any effect because you are selling to all countries.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Allowed countries rule card notice tests renders correctly when countries except some are allowed, others will be blocked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
      >
        <div
          class="fraud-protection-rule-card-notice-icon"
        >
          <svg
            data-testid="rule-card-notice-info-icon-svg"
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="9"
              r="5"
              stroke="#007CBA"
              stroke-width="1.42857"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="9"
              x2="15"
              y1="16.8818"
              y2="16.8818"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="10"
              x2="14"
              y1="19.25"
              y2="19.25"
            />
          </svg>
        </div>
        <div
          class="fraud-protection-rule-card-notice-text"
        >
          Orders from the following countries will be blocked by the filter: 
          <strong>
            Canada, United States
          </strong>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
    >
      <div
        class="fraud-protection-rule-card-notice-icon"
      >
        <svg
          data-testid="rule-card-notice-info-icon-svg"
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="9"
            r="5"
            stroke="#007CBA"
            stroke-width="1.42857"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="9"
            x2="15"
            y1="16.8818"
            y2="16.8818"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="10"
            x2="14"
            y1="19.25"
            y2="19.25"
          />
        </svg>
      </div>
      <div
        class="fraud-protection-rule-card-notice-text"
      >
        Orders from the following countries will be blocked by the filter: 
        <strong>
          Canada, United States
        </strong>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Allowed countries rule card notice tests renders correctly when countries except some are allowed, others will be hold 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
      >
        <div
          class="fraud-protection-rule-card-notice-icon"
        >
          <svg
            data-testid="rule-card-notice-info-icon-svg"
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="9"
              r="5"
              stroke="#007CBA"
              stroke-width="1.42857"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="9"
              x2="15"
              y1="16.8818"
              y2="16.8818"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="10"
              x2="14"
              y1="19.25"
              y2="19.25"
            />
          </svg>
        </div>
        <div
          class="fraud-protection-rule-card-notice-text"
        >
          Orders from the following countries will be screened by the filter: 
          <strong>
            Canada, United States
          </strong>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
    >
      <div
        class="fraud-protection-rule-card-notice-icon"
      >
        <svg
          data-testid="rule-card-notice-info-icon-svg"
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="9"
            r="5"
            stroke="#007CBA"
            stroke-width="1.42857"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="9"
            x2="15"
            y1="16.8818"
            y2="16.8818"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="10"
            x2="14"
            y1="19.25"
            y2="19.25"
          />
        </svg>
      </div>
      <div
        class="fraud-protection-rule-card-notice-text"
      >
        Orders from the following countries will be screened by the filter: 
        <strong>
          Canada, United States
        </strong>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Allowed countries rule card notice tests renders correctly when specific countries are allowed, others will be blocked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
      >
        <div
          class="fraud-protection-rule-card-notice-icon"
        >
          <svg
            data-testid="rule-card-notice-info-icon-svg"
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="9"
              r="5"
              stroke="#007CBA"
              stroke-width="1.42857"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="9"
              x2="15"
              y1="16.8818"
              y2="16.8818"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="10"
              x2="14"
              y1="19.25"
              y2="19.25"
            />
          </svg>
        </div>
        <div
          class="fraud-protection-rule-card-notice-text"
        >
          Orders from outside of the following countries will be blocked by the filter: 
          <strong>
            Canada, United States
          </strong>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
    >
      <div
        class="fraud-protection-rule-card-notice-icon"
      >
        <svg
          data-testid="rule-card-notice-info-icon-svg"
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="9"
            r="5"
            stroke="#007CBA"
            stroke-width="1.42857"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="9"
            x2="15"
            y1="16.8818"
            y2="16.8818"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="10"
            x2="14"
            y1="19.25"
            y2="19.25"
          />
        </svg>
      </div>
      <div
        class="fraud-protection-rule-card-notice-text"
      >
        Orders from outside of the following countries will be blocked by the filter: 
        <strong>
          Canada, United States
        </strong>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Allowed countries rule card notice tests renders correctly when specific countries are allowed, others will be hold 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
      >
        <div
          class="fraud-protection-rule-card-notice-icon"
        >
          <svg
            data-testid="rule-card-notice-info-icon-svg"
            fill="none"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="9"
              r="5"
              stroke="#007CBA"
              stroke-width="1.42857"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="9"
              x2="15"
              y1="16.8818"
              y2="16.8818"
            />
            <line
              stroke="#007CBA"
              stroke-width="1.5"
              x1="10"
              x2="14"
              y1="19.25"
              y2="19.25"
            />
          </svg>
        </div>
        <div
          class="fraud-protection-rule-card-notice-text"
        >
          Orders from outside of the following countries will be screened by the filter: 
          <strong>
            Canada, United States
          </strong>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-info"
    >
      <div
        class="fraud-protection-rule-card-notice-icon"
      >
        <svg
          data-testid="rule-card-notice-info-icon-svg"
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="9"
            r="5"
            stroke="#007CBA"
            stroke-width="1.42857"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="9"
            x2="15"
            y1="16.8818"
            y2="16.8818"
          />
          <line
            stroke="#007CBA"
            stroke-width="1.5"
            x1="10"
            x2="14"
            y1="19.25"
            y2="19.25"
          />
        </svg>
      </div>
      <div
        class="fraud-protection-rule-card-notice-text"
      >
        Orders from outside of the following countries will be screened by the filter: 
        <strong>
          Canada, United States
        </strong>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
